src/features/embalm/stepContent/steps/CreateSarcophagus.tsx

```diff
@@ -1,19 +1,72 @@
 import { Button, Flex, Heading, Text } from '@chakra-ui/react';
 import { ProgressTracker } from '../components/ProgressTracker';
 import { ProgressTrackerStage } from '../components/ProgressTrackerStage';
-import { CreateSarcophagusStage, useCreateSarcophagus } from '../hooks/useCreateSarcophagus';
+import { useCreateSarcophagus } from '../hooks/useCreateSarcophagus/useCreateSarcophagus';
 import { useSarcophagusParameters } from '../hooks/useSarcophagusParameters';
 import { ReviewSarcophagus } from '../components/ReviewSarcophagus';
 import { SummaryErrorIcon } from '../components/SummaryErrorIcon';
+import { useAllowance } from '../../../../hooks/sarcoToken/useAllowance';
+import { BigNumber, ethers } from 'ethers';
+import { useEffect, useState } from 'react';
+import { CreateSarcophagusStage, defaultCreateSarcophagusStages } from '../utils/createSarcophagus';
+import { useContract, useSigner } from 'wagmi';
+import {
+  EmbalmerFacet__factory,
+  SarcoTokenMock__factory,
+} from '@sarcophagus-org/sarcophagus-v2-contracts';
+import { useNetworkConfig } from '../../../../lib/config';
 
 export function CreateSarcophagus() {
-  const { currentStage, handleCreate, stageError, createSarcophagusStages, hasApproved } = useCreateSarcophagus();
+  const { allowance } = useAllowance();
+  const [createSarcophagusStages, setCreateSarcophagusStages] = useState<Record<number, string>>(
+    defaultCreateSarcophagusStages
+  );
+
+  const networkConfig = useNetworkConfig();
+  const { data: signer } = useSigner();
+
+  const embalmerFacet = useContract({
+    address: networkConfig.diamondDeployAddress,
+    abi: EmbalmerFacet__factory.abi,
+    signerOrProvider: signer,
+  });
+
+  const sarcoToken = useContract({
+    address: networkConfig.sarcoTokenAddress,
+    abi: SarcoTokenMock__factory.abi,
+    signerOrProvider: signer,
+  });
```

Comment:
These could possibly be moved to the context. May possibly be worth it to have a specific context for all the contract facets.